tests/system/test_integration.py

```diff
@@ -60,8 +71,36 @@ def test_load_docs_with_template_and_add_error(self):
         self.load_docs_with_template(f, self.errors_url, 'error', 4)
         self.assert_no_logged_warnings()
 
+        # compare existing ES documents for errors with new ones
+        rs = self.es.search(index=self.index_name, body={
+            "query": {"term": {"processor.event": "error"}}})
+        assert rs['hits']['total'] == 4, "found {} documents".format(rs['count'])
+        approved = json.load(open('error.approved.json'))
+        self.check_docs(approved, rs['hits']['hits'], 'error')
+
         self.check_backend_error_sourcemap(count=4)
 
+    def check_docs(self, approved, received, doc_type):
+        for rec_entry in received:
+            checked = False
+            rec = rec_entry['_source']
+            rec_id = rec[doc_type]['id']
+            for appr_entry in approved:
+                appr = appr_entry['_source']
+                if rec_id == appr[doc_type]['id']:
+                    checked = True
+                    self.assert_docs(rec[doc_type], appr[doc_type])
+                    self.assert_docs(rec['context'], appr['context'])
+                    self.assert_docs(rec['@timestamp'], appr['@timestamp'])
+                    self.assert_docs(rec['processor'], appr['processor'])
```

Comment:
could we just compare the docs in full instead of the individual keys `context`, `@timestamp` and `processor` ? for example, what about `marks` ?

Reply:
`marks` are part of `context` 
comparing the docs in full does not work as e.g. `_id` is transient.